src/qml/components/blockclockdial.cpp

```diff
@@ -139,6 +187,19 @@ void BlockClockDial::paintProgress(QPainter * painter)
     painter->drawArc(bounds, startAngle * 16, spanAngle * 16);
 }
 
+void BlockClockDial::paintConnectingAnimation(QPainter * painter)
+{
+    QPen pen;
+    pen.setWidthF(4);
+    setupConnectingGradient(pen);
+    pen.setBrush(QBrush(m_connecting_gradient));
+    pen.setCapStyle(Qt::RoundCap);
+    const QRectF bounds = getBoundsForPen(pen);
+    painter->setPen(pen);
+    painter->drawArc(bounds, m_connecting_start_angle * 16, m_connecting_end_angle * 16);
+    m_connecting_start_angle = decrementGradientAngle(m_connecting_start_angle);
+}
+
 void BlockClockDial::paintBackground(QPainter * painter)
```

Comment:
one thing we can lump in here, since this is introducing the idea of connected into this component, is some logic to only draw the background when connecting. Right now the dial can draw blocks or some of the progress while it is connected.

```diff 
diff --git a/src/qml/components/blockclockdial.cpp b/src/qml/components/blockclockdial.cpp
index 07f2af85d..63e83ff4d 100644
--- a/src/qml/components/blockclockdial.cpp
+++ b/src/qml/components/blockclockdial.cpp
@@ -252,9 +252,9 @@ void BlockClockDial::paint(QPainter * painter)
         return;
     }

-    if (synced()) {
+    if (synced() && connected()) {
         paintBlocks(painter);
-    } else {
+    } else if (!synced() && connected()) {
         paintProgress(painter);
     }
 }
```

Reply:
This is what I'm referring to on current master

| connecting | syncing |
| ---------- | ------- |
| <img width="752" alt="Screen Shot 2023-03-01 at 5 00 47 PM" src="https://user-images.githubusercontent.com/23396902/222274972-62600569-566f-44fe-b09a-ae851e91d9ee.png"> | <img width="752" alt="Screen Shot 2023-03-01 at 5 00 50 PM" src="https://user-images.githubusercontent.com/23396902/222275003-5a6b0dee-3c52-41ba-bb94-7d5ca98dde49.png"> |